### PR TITLE
DHFPROD-6545: Create commoncrawl performance project

### PR DESCRIPTION
**Pull request by devinnicholson**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: examples/commoncrawl/build.gradle
Comment (rjrudin) [inline examples/commoncrawl/build.gradle:23]: Since we're not compiling any Java code, you can remove the "java" plugin on line 19 and these repositories and dependencies blocks.

File: examples/commoncrawl/flows/mapping.flow.json
Comment (rjrudin) [inline examples/commoncrawl/flows/mapping.flow.json:1]: I like for these to have as little cruft as possible so they're easier to understand - can remove description, stopOnError, options, versions; and then from the step - additionalCollections, validateEntity, sourceCollection, customHook, retryLimit, batchSize, threadCount

File: examples/commoncrawl/flows/mastering.flow.json
Comment (rjrudin) [inline examples/commoncrawl/flows/mastering.flow.json:1]: Can remove the same properties from this as the mapping step.

File: examples/commoncrawl/gradle-dhs.properties
Comment (rjrudin) [inline examples/commoncrawl/gradle-dhs.properties:1]: We don't need this file - if someone wants to connect to a DHS instance, they'll pass in hubDhs=true and hubSsl=true .

File: examples/commoncrawl/src/main/entity-config/databases/final-database.json
Comment (rjrudin) [inline examples/commoncrawl/src/main/entity-config/databases/final-database.json:1]: We want to gitignore src/main/entity-config; everything is code-generated in this directory

File: examples/commoncrawl/src/main/ml-config/database-fields/final-database.xml
Comment (rjrudin) [inline examples/commoncrawl/src/main/ml-config/database-fields/final-database.xml:1]: We want to gitignore src/main/ml-config as well, unless we get to a point where we need to add a non-generated file to the directory

File: examples/commoncrawl/src/main/ml-config/entities.layout.json
Comment (rjrudin) [inline examples/commoncrawl/src/main/ml-config/entities.layout.json:1]: This file is specific to QuickStart and can be deleted

File: examples/commoncrawl/src/main/ml-config/security/users/exporter.json
Comment (rjrudin) [inline examples/commoncrawl/src/main/ml-config/security/users/exporter.json:1]: Do you know if this user is needed by any scripts? If not, let's delete it for now. If we need to keep it, we'll need to gitignore every directory under src/main/ml-config except this one.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Good start - there are some files we can delete/gitignore, and then I'd like to convert as many of the shell scripts into Gradle tasks as possible.

Also, can you run "gradlew hubConvert" on this project? It will convert the flows/steps into the HC model that allows this project to be used in HC.

File: examples/commoncrawl/src/main/ml-config/security/users/exporter.json
Comment (devinnicholson) [reply examples/commoncrawl/src/main/ml-config/security/users/exporter.json:1]: This user performs a perf test on nightly for CSV export

Comment (devinnicholson) [reply examples/commoncrawl/src/main/ml-config/security/users/exporter.json:1]: So, no, I guess we technically don't need it for examples proj

**Review by rjrudin — COMMENTED** [review]
I'll try to check this out locally soon, tied up right now with the NiFi release.

File: examples/commoncrawl/run_all_tests.sh
Comment (SameeraPriyathamTadikonda) [inline examples/commoncrawl/run_all_tests.sh:42]: We should not copy the credentials

File: examples/commoncrawl/build.gradle
Comment (SameeraPriyathamTadikonda) [inline examples/commoncrawl/build.gradle:57]: It's better to copy sample data to the example project. This location is not accessible to everyone.

File: examples/commoncrawl/run_all_tests.sh
Comment (devinnicholson) [reply examples/commoncrawl/run_all_tests.sh:42]: I'll msg you on Slack w/ some questions I have. This is PEAR credentials.

File: performance-tests/commoncrawl-project/gradle.properties
Comment (SameeraPriyathamTadikonda) [inline performance-tests/commoncrawl-project/gradle.properties:10]: Do we still need this? if so we might need to update the version.

File: performance-tests/commoncrawl-project/collect-stats.sh
Comment (rjrudin) [inline performance-tests/commoncrawl-project/collect-stats.sh:1]: I am thinking that since we have no need for this file for local dev/testing, perhaps it can be added to the project when it's used in the environment for doing real performance testing. Since we don't need it yet for the immediate goal of local dev/testing, let's remove it.

File: performance-tests/commoncrawl-project/converted-entities/Location.entity.json
Comment (rjrudin) [inline performance-tests/commoncrawl-project/converted-entities/Location.entity.json:11]: Instead of using elementRangeIndex or rangeIndex, add "facetable": true to these property definitions.

File: performance-tests/commoncrawl-project/converted-flows/flows/mapping.flow.json
Comment (rjrudin) [inline performance-tests/commoncrawl-project/converted-flows/flows/mapping.flow.json:1]: This directory can be safely deleted too.

File: performance-tests/commoncrawl-project/converted-entities/Location.entity.json
Comment (rjrudin) [reply performance-tests/commoncrawl-project/converted-entities/Location.entity.json:11]: Oops, just realized this is the converted-entities directory - you can safely delete this directory

File: performance-tests/commoncrawl-project/gradle.properties
Comment (rjrudin) [reply performance-tests/commoncrawl-project/gradle.properties:10]: This should be removed

File: performance-tests/commoncrawl-project/gradle.properties
Comment (rjrudin) [inline performance-tests/commoncrawl-project/gradle.properties:14]: Remove everything else in this file except for the *Password properties at the end for flow-developer and flow-operator - it's all boilerplate that we don't need, DHF knows all these default values. 

mlUsername and mlPassword should then be defined in gradle-local.properties

File: performance-tests/commoncrawl-project/steps/mapping/MapWat.step.json
Comment (rjrudin) [inline performance-tests/commoncrawl-project/steps/mapping/MapWat.step.json:17]: This is a pretty complex mapping step with all these custom functions - we'll likely slim this down to something with no mapping functions in the future, but it's fine for now. We'll definitely want a complex one like this too.

**Review by rjrudin — COMMENTED** [review]
Looks good! Just a few comments

**Comment by SameeraPriyathamTadikonda** [comment]
Since it is an example project and we don't need to build and run tests on this PR, Converting this is to draft.

**Comment by SameeraPriyathamTadikonda** [comment]
@rjrudin can we merge this PR?

**Review by rjrudin — APPROVED** [review]
@SameeraPriyathamTadikonda  @devinnicholson  Approving - I apologize, I have not had a chance to test this out locally yet, but the file content looks good for a starting point

**Comment by SameeraPriyathamTadikonda** [comment]
> @SameeraPriyathamTadikonda @devinnicholson Approving - I apologize, I have not had a chance to test this out locally yet, but the file content looks good for a starting point

Thank you @rjrudin. I'll merge this PR manually once other approved PRs are merged.

**Review by SameeraPriyathamTadikonda — APPROVED** [review]

**Pull request merged**
